engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,574 @@
+      return equalsNode.execute(selfMetaName, otherMetaName);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "!selfInterop.isDate(selfTimeZone)",
+      "!selfInterop.isTime(selfTimeZone)",
+      "selfInterop.isTimeZone(selfTimeZone)",
+      "!otherInterop.isDate(otherTimeZone)",
+      "!otherInterop.isTime(otherTimeZone)",
+      "otherInterop.isTimeZone(otherTimeZone)"
+  }, limit = "3")
+  boolean equalsTimeZones(Object selfTimeZone, Object otherTimeZone,
+      @CachedLibrary("selfTimeZone") InteropLibrary selfInterop,
+      @CachedLibrary("otherTimeZone") InteropLibrary otherInterop) {
+    try {
+      return selfInterop.asTimeZone(selfTimeZone).equals(
+          otherInterop.asTimeZone(otherTimeZone)
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @TruffleBoundary
+  @Specialization(guards = {
+      "selfInterop.isDate(selfZonedDateTime)",
+      "selfInterop.isTime(selfZonedDateTime)",
+      "selfInterop.isTimeZone(selfZonedDateTime)",
+      "otherInterop.isDate(otherZonedDateTime)",
+      "otherInterop.isTime(otherZonedDateTime)",
+      "otherInterop.isTimeZone(otherZonedDateTime)"
+  }, limit = "3")
+  boolean equalsZonedDateTimes(Object selfZonedDateTime, Object otherZonedDateTime,
+      @CachedLibrary("selfZonedDateTime") InteropLibrary selfInterop,
+      @CachedLibrary("otherZonedDateTime") InteropLibrary otherInterop) {
+    try {
+      return ZonedDateTime.of(
+          selfInterop.asDate(selfZonedDateTime),
+          selfInterop.asTime(selfZonedDateTime),
+          selfInterop.asTimeZone(selfZonedDateTime)
+      ).isEqual(
+          ZonedDateTime.of(
+              otherInterop.asDate(otherZonedDateTime),
+              otherInterop.asTime(otherZonedDateTime),
+              otherInterop.asTimeZone(otherZonedDateTime)
+          )
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isDate(selfDateTime)",
+      "selfInterop.isTime(selfDateTime)",
+      "!selfInterop.isTimeZone(selfDateTime)",
+      "otherInterop.isDate(otherDateTime)",
+      "otherInterop.isTime(otherDateTime)",
+      "!otherInterop.isTimeZone(otherDateTime)"
+  }, limit = "3")
+  boolean equalsDateTimes(Object selfDateTime, Object otherDateTime,
+      @CachedLibrary("selfDateTime") InteropLibrary selfInterop,
+      @CachedLibrary("otherDateTime") InteropLibrary otherInterop) {
+    try {
+      return LocalDateTime.of(
+          selfInterop.asDate(selfDateTime),
+          selfInterop.asTime(selfDateTime)
+      ).isEqual(
+          LocalDateTime.of(
+              otherInterop.asDate(otherDateTime),
+              otherInterop.asTime(otherDateTime)
+          )
+      );
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
```

Comment:
Btw. why do we have a specialization for local date-time? Do we have those anywhere in Enso or relevant parts of interop? I was sure that we just have ZonedDateTime.

Reply:
This is for interop values. We do not have local date-times without time zones in Enso.

> relevant parts of interop

I cannot point you to relevant parts of interop right now. We do not have explicit tests for that yet.